<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ParametersDialog</class>
 <widget class="QDialog" name="ParametersDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>300</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Parameters</string>
  </property>
  <property name="windowIcon">
   <iconset resource="resources.qrc">
    <normaloff>:/assets/i_params.png</normaloff>:/assets/i_params.png</iconset>
  </property>
  <layout class="QGridLayout" name="gridLayout_2">
   <item row="1" column="0">
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
      <widget class="QTabWidget" name="tabWidget">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>280</height>
        </size>
       </property>
       <property name="currentIndex">
        <number>1</number>
       </property>
       <widget class="QWidget" name="tab">
        <attribute name="title">
         <string>Parameters</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="2" column="0">
          <spacer name="verticalSpacer_4">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="0" column="0">
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <spacer name="verticalSpacer_3">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_8">
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_2">
               <item>
                <widget class="QLabel" name="label">
                 <property name="text">
                  <string>Universe size</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="warning_label">
                 <property name="enabled">
                  <bool>true</bool>
                 </property>
                 <property name="text">
                  <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; color:#ff0000;&quot;&gt;Game must be clear before&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                 </property>
                 <property name="textFormat">
                  <enum>Qt::RichText</enum>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QPushButton" name="reset_universe_pushButton">
               <property name="maximumSize">
                <size>
                 <width>25</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="toolTip">
                <string>Reset Universe</string>
               </property>
               <property name="text">
                <string/>
               </property>
               <property name="icon">
                <iconset resource="resources.qrc">
                 <normaloff>:/assets/i_reset.png</normaloff>:/assets/i_reset.png</iconset>
               </property>
               <property name="autoDefault">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="universe_spinBox">
               <property name="suffix">
                <string> cells</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <spacer name="verticalSpacer_2">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_7">
             <item>
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>Timer</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="reset_timer_pushButton">
               <property name="maximumSize">
                <size>
                 <width>25</width>
                 <height>25</height>
                </size>
               </property>
               <property name="toolTip">
                <string>Reset Timer</string>
               </property>
               <property name="text">
                <string/>
               </property>
               <property name="icon">
                <iconset resource="resources.qrc">
                 <normaloff>:/assets/i_reset_timer.png</normaloff>:/assets/i_reset_timer.png</iconset>
               </property>
               <property name="autoDefault">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="timer_spinBox">
               <property name="suffix">
                <string> ms</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_6">
             <item>
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>Color</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="reset_color_pushButton">
               <property name="maximumSize">
                <size>
                 <width>25</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="toolTip">
                <string>Reset Color</string>
               </property>
               <property name="text">
                <string/>
               </property>
               <property name="icon">
                <iconset resource="resources.qrc">
                 <normaloff>:/assets/i_reset_color.png</normaloff>:/assets/i_reset_color.png</iconset>
               </property>
               <property name="autoDefault">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="colorpushButton">
               <property name="minimumSize">
                <size>
                 <width>157</width>
                 <height>16</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>16</width>
                 <height>16</height>
                </size>
               </property>
               <property name="text">
                <string/>
               </property>
               <property name="iconSize">
                <size>
                 <width>175</width>
                 <height>16</height>
                </size>
               </property>
               <property name="autoDefault">
                <bool>false</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item row="3" column="0">
          <widget class="QPushButton" name="reset_parameter_pushButton">
           <property name="text">
            <string>Reset</string>
           </property>
           <property name="autoDefault">
            <bool>false</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="tab_2">
        <attribute name="title">
         <string>Game Mode</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout_7">
         <item row="2" column="0">
          <widget class="QPushButton" name="reset_mode_pushButton">
           <property name="text">
            <string>Reset</string>
           </property>
           <property name="autoDefault">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <layout class="QGridLayout" name="gridLayout_6">
           <item row="2" column="0">
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <widget class="QCheckBox" name="born_checkBox">
               <property name="text">
                <string>Born</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
               <property name="tristate">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="born_min_spinBox">
               <property name="suffix">
                <string> neighbour</string>
               </property>
               <property name="maximum">
                <number>8</number>
               </property>
               <property name="value">
                <number>3</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_5">
               <property name="maximumSize">
                <size>
                 <width>10</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>to</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="born_max_spinBox">
               <property name="suffix">
                <string>  neighbour</string>
               </property>
               <property name="maximum">
                <number>8</number>
               </property>
               <property name="value">
                <number>3</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="4" column="0">
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QCheckBox" name="stase_checkBox">
               <property name="text">
                <string>Stase</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="stase_min_spinBox">
               <property name="suffix">
                <string> neighbour</string>
               </property>
               <property name="maximum">
                <number>8</number>
               </property>
               <property name="value">
                <number>2</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_6">
               <property name="minimumSize">
                <size>
                 <width>10</width>
                 <height>0</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>10</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>to</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="stase_max_spinBox">
               <property name="suffix">
                <string> neighbour</string>
               </property>
               <property name="maximum">
                <number>8</number>
               </property>
               <property name="value">
                <number>2</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="3" column="0">
            <spacer name="verticalSpacer_10">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="0" column="0">
            <spacer name="verticalSpacer_11">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item row="1" column="0">
          <spacer name="verticalSpacer_9">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="tab_3">
        <attribute name="title">
         <string>Demo</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout_5">
         <item row="0" column="0">
          <layout class="QGridLayout" name="gridLayout_4">
           <item row="0" column="0">
            <layout class="QVBoxLayout" name="verticalLayout_3">
             <item>
              <spacer name="verticalSpacer_8">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>40</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QLabel" name="label_4">
               <property name="maximumSize">
                <size>
                 <width>16777215</width>
                 <height>112</height>
                </size>
               </property>
               <property name="toolTip">
                <string>https://en.wikipedia.org/wiki/Conway%27s_Game_of_Life#Examples_of_patterns</string>
               </property>
               <property name="text">
                <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;a href=&quot;https://en.wikipedia.org/wiki/Conway%27s_Game_of_Life#Examples_of_patterns&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;Example of patterns&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
               <property name="openExternalLinks">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="verticalSpacer_7">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>40</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_4">
               <item>
                <widget class="QSpinBox" name="random_spinBox">
                 <property name="suffix">
                  <string> % </string>
                 </property>
                 <property name="maximum">
                  <number>100</number>
                 </property>
                 <property name="value">
                  <number>50</number>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="random_pushButton">
                 <property name="text">
                  <string>Random </string>
                 </property>
                 <property name="autoDefault">
                  <bool>false</bool>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <spacer name="verticalSpacer_5">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>40</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout">
               <item>
                <widget class="QComboBox" name="demo_comboBox">
                 <item>
                  <property name="text">
                   <string>Aircraft carrier (stable)</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Beehive (stable)</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Boat (stable)</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Loaf (stable)</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Beacon (oscillator)</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Toad (oscillator)</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Pentadecathlon (oscillator)</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Glider (ship)</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Small ship (ship)</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Glider Gun (canon)</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>P46 Glider Gun (canon)</string>
                  </property>
                 </item>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="demo_load_pushButton">
                 <property name="text">
                  <string>Load demo</string>
                 </property>
                 <property name="autoDefault">
                  <bool>false</bool>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item row="1" column="0">
          <spacer name="verticalSpacer_6">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections/>
</ui>
